pkg/apis/controlplane/types.go

```diff
@@ -436,3 +436,64 @@ type EgressGroupList struct {
 	metav1.ListMeta
 	Items []EgressGroup
 }
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type SupportBundleCollection struct {
+	metav1.TypeMeta
+	metav1.ObjectMeta
+	ExpiredAt      metav1.Time
+	SinceTime      string
+	FileServer     BundleFileServer
+	Authentication BundleServerAuthConfiguration
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// SupportBundleCollectionList is a list of SupportBundleCollection objects.
+type SupportBundleCollectionList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+	Items           []SupportBundleCollection `json:"items" protobuf:"bytes,2,rep,name=items"`
+}
+
+type BundleFileServer struct {
+	// The URL of the file server. It is set with format: scheme://host[:port][/path],
+	// e.g, https://api.example.com:8443/v1/supportbundles/. If scheme is not set, https is used by default.
+	URL string
+}
+
+type BasicAuthentication struct {
+	Username string
+	Password string
+}
+
+type BundleServerAuthConfiguration struct {
```

Comment:
Ditto

Reply:
I don't think we can reuse CRD definition for `BundleServerAuthConfiguration`, because a Secret reference is used in the CRD definition for the authenticator, while in the control plane API, the field is configured with the actual authenticator parsed from the Secret by Controller.